### PR TITLE
Added an example of per-project dependency management tool. Fixes #162.

### LINKED ISSUE #162: Example per-project library collections
This may not be needed but I could use an example to go with this sentence because I couldn't think of one on my own.

"Even languages that don’t explicitly have virtual environments have other mechanisms that promote per-project library collections." in [section 1 - virtual environments episode](https://carpentries-incubator.github.io/python-intermediate-development/12-virtual-environments/index.html)

### PR DESCRIPTION
**Pull request by anenadic**

### CODE REVIEW

File: _episodes/12-virtual-environments.md [hunk 1]
```diff
@@ -63,10 +63,10 @@ which contains a particular version of Python installation
 plus a number of additional external libraries.
 
 Virtual environments are not just a feature of Python -
-most modern programming languages use them to isolate libraries for a specific project
+most modern programming languages use them to isolate libraries/dependencies for a specific project
```
Comment (bielsnohr) [inline _episodes/12-virtual-environments.md:66]:
```suggestion
most modern programming languages use them to isolate libraries or dependencies for a specific project
```

File: _episodes/12-virtual-environments.md [hunk 2]
```diff
@@ -63,10 +63,10 @@ which contains a particular version of Python installation
...
 and make it easier to develop, run, test and share code with others.
-Even languages that don't explicitly have virtual environments have other mechanisms
-that promote per-project library collections.
+Even languages that do not explicitly have virtual environments have mechanisms
+that promote per-project dependency collections (e.g. Maven with classpath for Java).
```
Comment (bielsnohr) [inline _episodes/12-virtual-environments.md:69]: Might be nice to give some more examples here? I think I included more in the slides that I use, but just using those from the linked issue #162 : 

```suggestion
that promote per-project dependency collections:

- Maven with classpath for Java
- Bundler for Ruby
- Conan and per-project environments for C++
- Spack, used extensively for package management in HPC contexts, is a more generic, language agnostic tool that also has something resembling virtual environments
```

**Review by bielsnohr — CHANGES_REQUESTED** [review]
Just a few minor changes to consider.

File: _episodes/12-virtual-environments.md (hunk 2 above)
Comment (anenadic) [reply _episodes/12-virtual-environments.md:69]: Do not know about Conan and Spack but Bundler for ruby is actually a tool that enables creation of virtual environments similar to venv (locking the ruby and gem versions) and we are trying to give examples of tools that are not explicitly doing that but are providing some kind of dependency management. For that reason I did not include any of the above even though they were mentioned in the issue. Perhaps a good example does not exist? Maybe we just need to get rid of that sentence: "Even languages that don't explicitly have virtual environments have other mechanisms that promote per-project library collections." @bielsnohr ?

Comment (bielsnohr) [reply _episodes/12-virtual-environments.md:69]: I think the confusing bit is that most languages combine the tooling that does package management and dependency isolation, whereas in Python they are distinct tools that can work together (pip and venv). I agree that we should probably just get rid of the sentence

> Even languages that do not explicitly have virtual environments have mechanisms that promote per-project dependency collections (e.g. Maven with classpath for Java).

"Virtual environments" is a term very specific to Python, and it doesn't really apply well to other languages. I'll suggest an alternate wording in another review comment.

File: _episodes/12-virtual-environments.md [hunk 3]
```diff
@@ -63,7 +63,7 @@ which contains a particular version of Python installation
 plus a number of additional external libraries.
 
 Virtual environments are not just a feature of Python -
```
Comment (bielsnohr) [inline _episodes/12-virtual-environments.md:65]:
```suggestion
Virtual environments are not just a feature of Python---
```

Comment (anenadic) [reply _episodes/12-virtual-environments.md:65]: I like this proposal - think it explains things more clearly - thank you @bielsnohr.

File: _episodes/12-virtual-environments.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ which contains a particular version of Python installation
...
-most modern programming languages use them to isolate libraries/dependencies for a specific project
+most modern programming languages use them to isolate libraries or dependencies for a specific project
 and make it easier to develop, run, test and share code with others.
 Even languages that do not explicitly have virtual environments have mechanisms
 that promote per-project dependency collections (e.g. Maven with classpath for Java).
```
[the remark this answers is not in the document]
Comment (anenadic) [reply _episodes/12-virtual-environments.md]: Accepted changes - you still need to approve the PR @bielsnohr.

**Review by bielsnohr — APPROVED** [review]

**Pull request merged**